### PR TITLE
Don't create a constant node for Vector64

### LINKED ISSUE #105537: [mono] Vector shuffle failures in Mono LLVM scenario
## Build Information
Build: https://dev.azure.com/dnceng-public/public/_build/results?buildId=755747
Build error leg or test failing: `Microsoft.Bcl.Memory.Tests`, `System.Runtime.Intrinsics.Tests `, `System.Security.Cryptography.Tests`
Affected CI: `tvos_arm64_release_allsubsets_mono` (`runtime-extra-platforms`). Note: x64 LLVM coverage is not working currently due to https://github.com/dotnet/runtime/issues/105319
Range of commits: https://github.com/dotnet/runtime/compare/d73e625c...7134e1e6

Full logs: 
[net.dot.System.Security.Cryptography.Tests.log](https://github.com/user-attachments/files/16386575/net.dot.System.Security.Cryptography.Tests.log)
[net.dot.System.Runtime.Intrinsics.Tests.log](https://github.com/user-attachments/files/16386576/net.dot.System.Runtime.Intrinsics.Tests.log)
[net.dot.Microsoft.Bcl.Memory.Tests.log](https://github.com/user-attachments/files/16386577/net.dot.Microsoft.Bcl.Memory.Tests.log)

Stack trace example for `System.Runtime.Intrinsics.Tests`:
```
[17:54:10.7789630] 2024-07-25 14:54:10.805 System.Runtime.Intrinsics.Tests[3591:104610303] 1) 	[FAIL] System.Runtime.Intrinsics.Tests.Vectors.Vector64Tests.Vector64SByteShuffleOneInputWithDirectVectorTest   Test name: System.Runtime.Intrinsics.Tests.Vectors.Vector64Tests.Vector64SByteShuffleOneInputWithDirectVectorTest
[17:54:10.7789730] 2024-07-25 14:54:10.806 System.Runtime.Intrinsics.Tests[3591:104610303]    Assembly:  [System.Runtime.Intrinsics.Tests, Version=9.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51]
[17:54:10.7789800]    Exception messages: Assert.Equal() Failure: Values differ
[17:54:10.7789860] Expected: 8
[17:54:10.7789940] 2024-07-25 14:54:10.806 System.Runtime.Intrinsics.Tests[3591:104610303] Actual:   0   Exception stack traces:    at System.Runtime.Intrinsics.Tests.Vectors.Vector64Tests.Vector64SByteShuffleOneInputWithDirectVectorTest()
[17:54:10.7790010]    at System.Reflection.MethodBaseInvoker.InterpretedInvoke_Method(Object obj, IntPtr* args)
[17:54:10.7790120]    at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
[17:54:10.7790190] 2024-07-25 14:54:10.806 System.Runtime.Intrinsics.Tests[3591:104610303]    Execution time: 0.0073298
[17:54:10.7790280] 2024-07-25 14:54:10.806 System.Runtime.Intrinsics.Tests[3591:104610303] 2) 	[FAIL] System.Runtime.Intrinsics.Tests.Vectors.Vector64Tests.Vector64SingleCreateFromArrayTest   Test name: System.Runtime.Intrinsics.Tests.Vectors.Vector64Tests.Vector64SingleCreateFromArrayTest
[17:54:10.7806070] 2024-07-25 14:54:10.808 System.Runtime.Intrinsics.Tests[3591:104610303]    Assembly:  [System.Runtime.Intrinsics.Tests, Version=9.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51]
[17:54:10.7806290]    Exception messages: Assert.Equal() Failure: Values differ
[17:54:10.7806370] Expected: <0, 0>
[17:54:10.7826440] 2024-07-25 14:54:10.809 System.Runtime.Intrinsics.Tests[3591:104610303] Actual:   <1, 2>   Exception stack traces:    at System.Runtime.Intrinsics.Tests.Vectors.Vector64Tests.Vector64SingleCreateFromArrayTest()
[17:54:10.7826640]    at System.Reflection.MethodBaseInvoker.InterpretedInvoke_Method(Object obj, IntPtr* args)
[17:54:10.7826730]    at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
[17:54:10.7826790]    Execution time: 0.0058828
[17:54:10.7827190] 2024-07-25 14:54:10.810 System.Runtime.Intrinsics.Tests[3591:104610303] 3) 	[FAIL] System.Runtime.Intrinsics.Tests.Vectors.Vector64Tests.Vector64UInt32ShuffleOneInputWithDirectVectorTest   Test name: System.Runtime.Intrinsics.Tests.Vectors.Vector64Tests.Vector64UInt32ShuffleOneInputWithDirectVectorTest
[17:54:10.7827320] 2024-07-25 14:54:10.810 System.Runtime.Intrinsics.Tests[3591:104610303]    Assembly:  [System.Runtime.Intrinsics.Tests, Version=9.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51]
[17:54:10.7827390]    Exception messages: Assert.Equal() Failure: Values differ
[17:54:10.7827440] Expected: 2
[17:54:10.7827520] 2024-07-25 14:54:10.810 System.Runtime.Intrinsics.Tests[3591:104610303] Actual:   0   Exception stack traces:    at System.Runtime.Intrinsics.Tests.Vectors.Vector64Tests.Vector64UInt32ShuffleOneInputWithDirectVectorTest()
[17:54:10.7827590]    at System.Reflection.MethodBaseInvoker.InterpretedInvoke_Method(Object obj, IntPtr* args)
[17:54:10.7827650]    at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
[17:54:10.7828010] 2024-07-25 14:54:10.810 System.Runtime.Intrinsics.Tests[3591:104610303]    Execution time: 0.0033379
[17:54:10.7828650] 2024-07-25 14:54:10.810 System.Runtime.Intrinsics.Tests[3591:104610303] 4) 	[FAIL] System.Runtime.Intrinsics.Tests.Vectors.Vector64Tests.Vector64Int16ShuffleOneInputTest   Test name: System.Runtime.Intrinsics.Tests.Vectors.Vector64Tests.Vector64Int16ShuffleOneInputTest
[17:54:10.7828830] 2024-07-25 14:54:10.810 System.Runtime.Intrinsics.Tests[3591:104610303]    Assembly:  [System.Runtime.Intrinsics.Tests, Version=9.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51]
[17:54:10.7828900]    Exception messages: Assert.Equal() Failure: Values differ
[17:54:10.7828950] Expected: 4
[17:54:10.7838870] 2024-07-25 14:54:10.811 System.Runtime.Intrinsics.Tests[3591:104610303] Actual:   0   Exception stack traces:    at System.Runtime.Intrinsics.Tests.Vectors.Vector64Tests.Vector64Int16ShuffleOneInputTest()
[17:54:10.7839030]    at System.Reflection.MethodBaseInvoker.InterpretedInvoke_Method(Object obj, IntPtr* args)
[17:54:10.7839100]    at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
[17:54:10.7839150]    Execution time: 0.0020901
[17:54:10.7854900] 2024-07-25 14:54:10.813 System.Runtime.Intrinsics.Tests[3591:104610303] 5) 	[FAIL] System.Runtime.Intrinsics.Tests.Vectors.Vector64Tests.Vector64ByteShuffleOneInputWithZeroIndicesTest   Test name: System.Runtime.Intrinsics.Tests.Vectors.Vector64Tests.Vector64ByteShuffleOneInputWithZeroIndicesTest
[17:54:10.7855200] 2024-07-25 14:54:10.813 System.Runtime.Intrinsics.Tests[3591:104610303]    Assembly:  [System.Runtime.Intrinsics.Tests, Version=9.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51]
[17:54:10.7855260]    Exception messages: Assert.Equal() Failure: Values differ
[17:54:10.7855300] Expected: 1
[17:54:10.7868500] 2024-07-25 14:54:10.814 System.Runtime.Intrinsics.Tests[3591:104610303] Actual:   0   Exception stack traces:    at System.Runtime.Intrinsics.Tests.Vectors.Vector64Tests.Vector64ByteShuffleOneInputWithZeroIndicesTest()
[17:54:10.7868640]    at System.Reflection.MethodBaseInvoker.InterpretedInvoke_Method(Object obj, IntPtr* args)
[17:54:10.7868700]    at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
[17:54:10.7868980] 2024-07-25 14:54:10.814 System.Runtime.Intrinsics.Tests[3591:104610303]    Execution time: 0.0036366
[17:54:10.7869600] 2024-07-25 14:54:10.814 System.Runtime.Intrinsics.Tests[3591:104610303] 6) 	[FAIL] System.Runtime.Intrinsics.Tests.Vectors.Vector64Tests.Vector64SByteShuffleOneInputWithZeroIndicesTest   Test name: System.Runtime.Intrinsics.Tests.Vectors.Vector64Tests.Vector64SByteShuffleOneInputWithZeroIndicesTest
[17:54:10.7870160] 2024-07-25 14:54:10.814 System.Runtime.Intrinsics.Tests[3591:104610303]    Assembly:  [System.Runtime.Intrinsics.Tests, Version=9.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51]
[17:54:10.7870310]    Exception messages: Assert.Equal() Failure: Values differ
[17:54:10.7870370] Expected: 1
[17:54:10.7870440] 2024-07-25 14:54:10.815 System.Runtime.Intrinsics.Tests[3591:104610303] Actual:   0   Exception stack traces:    at System.Runtime.Intrinsics.Tests.Vectors.Vector64Tests.Vector64SByteShuffleOneInputWithZeroIndicesTest()
[17:54:10.7870520]    at System.Reflection.MethodBaseInvoker.InterpretedInvoke_Method(Object obj, IntPtr* args)
[17:54:10.7870580]    at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
[17:54:10.7870810] 2024-07-25 14:54:10.815 System.Runtime.Intrinsics.Tests[3591:104610303]    Execution time: 0.0019761
```

## Error Message

Fill the error message using [step by step known issues guidance](https://github.com/dotnet/arcade/blob/main/Documentation/Projects/Build%20Analysis/KnownIssueJsonStepByStep.md).

```json
{
  "ErrorMessage": "[FAIL] System.Runtime.Intrinsics.Tests.Vectors.Vector64Tests.Vector64SByteShuffleOneInputWithDirectVectorTest   Test name: System.Runtime.Intrinsics.Tests.Vectors.Vector64Tests.Vector64SByteShuffleOneInputWithDirectVectorTest",
  "BuildRetry": false,
  "ExcludeConsoleLog": false
}
```

 ### Known issue validation
**Build: :mag_right:** https://dev.azure.com/dnceng-public/public/_build/results?buildId=755747
**Error message validated:** `[[FAIL] System.Runtime.Intrinsics.Tests.Vectors.Vector64Tests.Vector64SByteShuffleOneInputWithDirectVectorTest   Test name: System.Runtime.Intrinsics.Tests.Vectors.Vector64Tests.Vector64SByteShuffleOneInputWithDirectVectorTest`]
**Result validation:** :x: Known issue did not match with the provided build.
**Validation performed at:** 7/26/2024 4:35:54 AM UTC

#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|0|0|0|

### PR DESCRIPTION
**Pull request by tannergooding**
This fixes https://github.com/dotnet/runtime/issues/105537

### CODE REVIEW

**Comment by tannergooding** [comment]
@matouskozak, is there a way to kick of the job that failed for https://github.com/dotnet/runtime/issues/105537?

**Comment by matouskozak** [comment]
> @matouskozak, is there a way to kick of the job that failed for #105537?
> 
> _edit:_ Found the command, I had missed that this was just extra-platforms, I had misclicked and couldn't find the job name initially

Yes, extra-platforms is fine. You can also use just `runtime-ioslike` which will trigger the tvos-arm64 job that is failing.

**Comment by tannergooding** [comment]
Looks like there's a non-trivial number of failures in the regular runs that are independent of the break found in https://github.com/dotnet/runtime/issues/105537

Going to try and see if I can discern what's existing and what's new. Will end up putting a PR up to revert the change if this isn't trivial to resolve

**Comment by matouskozak** [comment]
> Looks like there's a non-trivial number of failures in the regular runs that are independent of the break found in #105537
> 
> Going to try and see if I can discern what's existing and what's new. Will end up putting a PR up to revert the change if this isn't trivial to resolve

I'll help with that after the run finishes. The mobile-related failures are tracked at https://github.com/dotnet/runtime/issues/103472 which should help with discerning what's related.

**Comment by matouskozak** [comment]
> > Looks like there's a non-trivial number of failures in the regular runs that are independent of the break found in #105537
> > Going to try and see if I can discern what's existing and what's new. Will end up putting a PR up to revert the change if this isn't trivial to resolve
> 
> I'll help with that after the run finishes. The mobile-related failures are tracked at #103472 which should help with discerning what's related.

I took a brief look and the `System.Runtime.Intrinsics.Tests` are no longer failing on tvos-arm64 job. However, the `Microsoft.Bcl.Memory.Tests` still do, [crash dump](https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-105538-merge-a487ca1bace44d9b9d/Microsoft.Bcl.Memory.Tests/1/net.dot.Microsoft.Bcl.Memory.Tests.log?helixlogtype=result). I think this failure could still be related, right?

**Comment by tannergooding** [comment]
> However, the Microsoft.Bcl.Memory.Tests still do, [crash dump](https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-105538-merge-a487ca1bace44d9b9d/Microsoft.Bcl.Memory.Tests/1/net.dot.Microsoft.Bcl.Memory.Tests.log?helixlogtype=result). I think this failure could still be related, right?

Could be as there weren't any changes to Base64UrlDecoder in the past few days, but `Base64Decoder` isn't actually using `Vector128.Shuffle` anywhere, notably. It's also a bit odd that only these tests are failing and only on this one platform.

The log doesn't appear to have any meaningful stack trace, all just indicating something fails in reflection as part of `InterpretedInvoke_Method`. `Base64Decoder` then uses a `SimdShuffle` helper that defers down to `Vector128.ShuffleUnsafe` which itself defers down to `AdvSimd.Arm64.VectorTableLookup`, so it "shouldn't" be hitting this code path anywhere.

Is there an easy way to get additional information/context for the failure here?

**Comment by steveisok** [comment]
> Is there an easy way to get additional information/context for the failure here?

Is there an actual crash in `Microsoft.Bcl.Memory.Tests` suite? I don't see it from the logs. 

Given there are individual test failures, I would recommend running it local on an ios device and see if you can and debug. If you don't have a device, I'm sure @matouskozak could help. I don't think the failure is exclusive to tvOS (we only run a subset on ios).

**Comment by tannergooding** [comment]
> Is there an actual crash in Microsoft.Bcl.Memory.Tests suite? I don't see it from the logs.

I don't either, just the 8 test failures failing from an xunit assert.

>  If you don't have a device, I'm sure @matouskozak could help.

I don't have such a device, unfortunately. So any help getting a better log, disassembly, LLVM IR, etc would be appreciated

**Review by fanyang-mono — APPROVED** [review]

File: src/mono/mono/mini/simd-intrinsics.c [hunk 1]
```diff
@@ -1352,10 +1351,8 @@ emit_vector_create_elementwise (
 	if (vector_size == 16) {
 		for (int i = 0; i < param_count; ++i) {
 			if (!is_const (args[i])) {
-				all_const = false;
-				break;
```
Comment (tannergooding) [inline src/mono/mono/mini/simd-intrinsics.c:1356]: This was the remaining issue.

Mono only supports 128-bit `xconst` today, but we're still trying to handle the case where only some inputs are constant; such as `Vector128.Create(1, 2, x, 4)`. This was early exiting in the case that not all inputs were constant so in the main loop below we ended up getting `Vector128.Create(1, 2, x, 0)` instead.

Fixed it to not early exit and only skip parameters if some were constant; which lets us get the correct value instead.

**Pull request merged**